service/src/main/java/tennisclub/service/TournamentService.java

```diff
@@ -0,0 +1,19 @@
+package tennisclub.service;
+
+import tennisclub.entity.Tournament;
+import tennisclub.entity.ranking.Ranking;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public interface TournamentService {
+    void create(Tournament tournament);
+    Tournament update(Tournament tournament);
+    void remove(Tournament tournament);
+    List<Tournament> listAll();
+    List<Tournament> listByStartTime(LocalDateTime startTime);
+    List<Tournament> listByEndTime(LocalDateTime endTime);
+    List<Tournament> listByTimeInterval(LocalDateTime from, LocalDateTime to);
+    List<Tournament> listByCapacity(Integer capacity);
+    void addRanking(Tournament tournament, Ranking ranking);
+}
```

Comment:
Same as for the Lesson service, I'm missing here a method findById() which you probably planned to add later :)

Reply:
Thanks! 😄